File: LeetCode/Java Solutions/Find The Original Array of Prefix Xor/find_xor.java
class Solution {
    public int[] findArray(int[] pref) {
        int n = pref.length;
        int[] arr = new int[n];
        arr[0] = pref[0];  // for i=0, arr[0] is same as pref[0]
        for (int i = 1; i < n; i++) {
            arr[i] = pref[i] ^ pref[i-1];
        }
        return arr;
    }
}

File: LeetCode/Python Solutions/Find The Original Array of Prefix Xor/find_xor.py
class Solution:
    def findArray(self, pref: 'List[int]') -> 'List[int]':
        n = len(pref)
        arr = [pref[0]]  # for i=0, arr[0] is same as pref[0]
        for i in range(1, n):
            arr.append(pref[i] ^ pref[i-1])
        return arr
